### PR TITLE
Fixed default binding behavior in scalars package

### LINKED ISSUE #4583: Bind UnsingedShort implicit
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

https://github.com/ChilliCream/hotchocolate/blob/main/src/HotChocolate/Core/src/Types.Scalars/UnsignedShortType.cs#L24

### Steps to reproduce

I think the type should be bound implicit

### Relevant log output

_No response_

### Additional Context?

_No response_

### Product

Hot Chocolate

### Version

12.4.0

### PR DESCRIPTION
**Pull request by nelsonprsousa**
Summary of the changes (Less than 80 chars)

- Bind SignedByteType, UnsignedIntType, UnsignedLongType, and UnsignedShortType implicit

Closes #4583

### CODE REVIEW

**Comment by michaelstaib** [comment]
You also need to write some unit tests for such a change.

**Review by michaelstaib — CHANGES_REQUESTED** [review]
We need tests for this to be merged.

**Comment by PascalSenn** [comment]
@nelsonprsousa Thanks for the PR :slightly_smiling_face:
Do you want to add a test or should i do this?
it’s basically this
```csharp
public class Query 
{
     public sbyte Byte {get;}
}
```
and
[Fact]
public async Task SByteType_Should_BeBoundImplicity_When_Registered()
{
    // arrange
    // act
    var executor = await new ServiceCollection()
            .AddGraphQL()
            .AddQueryType<Query>()
            .AddType<SByteType>() 
            .BuildRequestExecutorAsync();

    // assert
    executor.Schema.Print().MatchSnapshot();
}

**Comment by michaelstaib** [comment]
Well said @PascalSenn :D I was heads down in my own PRs and did not fully really explain and appreciate this :) 

We really appreciate and value any new contributor 💯  So thanks for your contribution. Pascal outlined the rest.

**Comment by nelsonprsousa** [comment]
> Well said @PascalSenn :D I was heads down in my own PRs and did not fully really explain and appreciate this :)
> 
> We really appreciate and value any new contributor 💯 So thanks for your contribution. Pascal outlined the rest.

No worries @michaelstaib 

Just pushed a new commit with 4 new unit tests. Also, we are now using file-scoped namespaces like it is specified by IDE0161.

cc @PascalSenn

File: src/HotChocolate/Core/test/Types.Scalars.Tests/ScalarTypeTestBase.cs [hunk 1]
```diff
@@ -1,232 +1,255 @@
...
+        where TType : ScalarType
+        where TDefaultClass : class
+    {
+        // arrange
+        // act
+        IRequestExecutor executor = await new ServiceCollection()
+                .AddGraphQL()
+                .AddQueryType<TDefaultClass>()
+                .AddType<TType>()
+                .BuildRequestExecutorAsync();
```
Comment (michaelstaib) [inline src/HotChocolate/Core/test/Types.Scalars.Tests/ScalarTypeTestBase.cs:245]:
```suggestion
        IRequestExecutor executor = await new ServiceCollection()
            .AddGraphQL()
            .AddQueryType<TDefaultClass>()
            .AddType<TType>()
            .BuildRequestExecutorAsync();
```

**Review by michaelstaib — APPROVED** [review]
@PascalSenn can you have another look and merge? So far it looks good to me.

File: src/HotChocolate/Core/test/Types.Scalars.Tests/ScalarTypeTestBase.cs [hunk 2]
```diff
@@ -1,232 +1,255 @@
 using System;
+using System.Threading.Tasks;
+
```
Comment (PascalSenn) [inline src/HotChocolate/Core/test/Types.Scalars.Tests/ScalarTypeTestBase.cs:3]: Can we remove those empty lines? :)

Comment (nelsonprsousa) [reply src/HotChocolate/Core/test/Types.Scalars.Tests/ScalarTypeTestBase.cs]: Done 👍

**Review by PascalSenn — APPROVED** [review]

File: src/HotChocolate/Core/test/Types.Scalars.Tests/ScalarTypeTestBase.cs (hunk 2 above)
Comment (PascalSenn) [reply src/HotChocolate/Core/test/Types.Scalars.Tests/ScalarTypeTestBase.cs]: awesome!

**Review by PascalSenn — APPROVED** [review]
Looks good! thanks for the contribution :)

**Pull request merged**